File: MoonEngine-Root/Assets/src/main/java/com/greentree/engine/moon/assets/location/AssetLocation.java
package com.greentree.engine.moon.assets.location;

import com.greentree.engine.moon.assets.key.AssetKey;

public interface AssetLocation {
	
	AssetKey getKey(String name);
	
	default boolean isExist(String name) {
		return getKey(name) != null;
	}
	
}

File: MoonEngine/Assets/src/main/java/com/greentree/engine/moon/assets/location/AssetLocation.kt
package com.greentree.engine.moon.assets.location

import com.greentree.engine.moon.assets.key.AssetKey
import java.util.*

interface AssetLocation {

	fun getKey(name: String): AssetKey = getKeyOptional(name).get()

	fun getKeyOptional(name: String): Optional<AssetKey>

	fun isExist(name: String) = getKeyOptional(name).isPresent
}